app/api/v2/resource/addresses.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+module API::V2
+  module Resource
+    # Addresses API
+    class Addresses < Grape::API
+      desc 'Documents related routes'
+      resource :addresses do
+        desc 'Upload a new address approval document for current user',
+             security: [{ 'BearerToken': [] }],
+             success: { code: 201, message: 'Document is uploaded' },
+             failure: [
+               { code: 400, message: 'Required params are empty' },
+               { code: 401, message: 'Invalid bearer token' },
+               { code: 422, message: 'Validation errors' }
+             ]
+        params do
+          requires :country,
+                   type: String,
+                   allow_blank: false,
+                   desc: 'Document type'
+          requires :address,
+                   type: String,
+                   allow_blank: false,
+                   desc: 'Document number'
+          requires :upload,
+                   desc: 'Array of Rack::Multipart::UploadedFile'
+          requires :city,
+                   allow_blank: true,
+                   desc: 'Document expiration date'
+          requires :postcode, type: String, desc: 'Any additional key: value pairs in json string format'
+        end
+
+        post do
+          identificator = SecureRandom.hex(16)
```

Comment:
We do not allow to create identificator here, because we dont want to link files from different requests.
But we DO on documents, because of different doc categories